modules/hal_infineon/mtb-hal-cat1/CMakeLists.txt

```diff
@@ -86,7 +86,7 @@ zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_TRNG    ${hal_dir}/source/cyhal
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_UART    ${hal_dir}/source/cyhal_uart.c)
 zephyr_library_sources_ifdef(CONFIG_USE_INFINEON_WDT     ${hal_dir}/source/cyhal_wdt.c)
 
-if(CONFIG_USE_INFINEON_ADC)
+if(CONFIG_USE_INFINEON_ADC OR CONFIG_USE_INFINEON_SMIF)
```

Comment:
This if statement is strange. Why do we need cyhal_analog_common.c when we use the SMIF?

Reply:
cyhal_analog_common.c is not required for SMIF, only DMA is required. Moved cyhal_analog_common.c out of this condition into a separate line